signer/src/testing/storage/model.rs

```diff
@@ -564,3 +567,57 @@ fn vec_diff<T: std::cmp::Eq + std::hash::Hash>(subtrahend: &mut Vec<T>, minuend:
     let minuend_set = minuend.iter().collect::<HashSet<_>>();
     subtrahend.retain(|v| !minuend_set.contains(v));
 }
+
+#[cfg(test)]
+mod tests {
+    use more_asserts::assert_ge;
+    use rand::SeedableRng as _;
+
+    use crate::{
+        storage::{self, DbRead as _},
+        testing,
+    };
+
+    use super::*;
+
+    #[tokio::test]
+    async fn check_simple_chain() {
+        let mut store = storage::in_memory::Store::new_shared();
+        let mut rng = rand::rngs::StdRng::seed_from_u64(42);
+
+        let test_model_params = Params {
+            num_bitcoin_blocks: 10,
+            num_stacks_blocks_per_bitcoin_block: 5,
+            num_deposit_requests_per_block: 0,
+            num_withdraw_requests_per_block: 0,
+            num_signers_per_request: 0,
+            consecutive_blocks: true,
+        };
+        let signer_set = testing::wsts::generate_signer_set_public_keys(&mut rng, 7);
+
+        let test_data = TestData::generate(&mut rng, &signer_set, &test_model_params);
+        test_data.write_to(&mut store).await;
+
+        let bitcoin_chain_tip = store
+            .get_bitcoin_canonical_chain_tip()
+            .await
+            .unwrap()
+            .unwrap();
+        let tip = store
+            .get_stacks_chain_tip(&bitcoin_chain_tip)
+            .await
+            .unwrap()
+            .unwrap();
+
+        let mut walk = vec![tip];
+        while let Some(current) = store
+            .get_stacks_block(&walk.last().unwrap().parent_hash)
+            .await
+            .unwrap()
+        {
+            walk.push(current);
+        }
+
+        assert_ge!(walk.len(), 10);
```

Comment:
Can we also check that the heights increment in order. Actually, I'm not sure that the stacks block heights are used so maybe we don't need it.